src/lib/device/journal_vdev.cpp

```diff
@@ -472,11 +472,19 @@ off_t JournalVirtualDev::Descriptor::dev_offset(off_t nbytes) const {
 }
 
 void JournalVirtualDev::Descriptor::update_data_start_offset(off_t offset) {
-    m_data_start_offset = offset;
-    auto data_start_offset_aligned = sisl::round_down(m_data_start_offset, m_vdev.info().chunk_size);
-    m_end_offset = data_start_offset_aligned + m_journal_chunks.size() * m_vdev.info().chunk_size;
-    LOGINFOMOD(journalvdev, "Updated data start offset off 0x{} {}", to_hex(offset), to_string());
-    RELEASE_ASSERT_EQ(m_end_offset - data_start_offset_aligned, m_total_size, "offset size mismatch {}", to_string());
+    if (!m_journal_chunks.empty()) {
+        m_data_start_offset = offset;
+        auto data_start_offset_aligned = sisl::round_down(m_data_start_offset, m_vdev.info().chunk_size);
+        m_end_offset = data_start_offset_aligned + m_journal_chunks.size() * m_vdev.info().chunk_size;
+        LOGINFOMOD(journalvdev, "Updated data start offset off 0x{} {}", to_hex(offset), to_string());
+        RELEASE_ASSERT_EQ(m_end_offset - data_start_offset_aligned, m_total_size, "offset size mismatch {}",
+                          to_string());
+    } else {
+        // If there are no chunks, we round up to the next chunk size.
```

Comment:
Can you explain a bit when this `else` branch will be hit?

Reply:
As discussed we round up.